.github/compatibility-test-matrices/compat.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python3
```

Comment:
Added an alternative in Python. Though bash does the trick, the script seemed like it was doing enough to warrant a language that doesn't make your eyes hurt.

I'm generally open to sticking with bash but find this solution both more readable, more maintainable and less error prone.

Still need to get some feedback from Carlos to make sure the script does what is typically required. Other than that, willing to drop this and go with bash if people prefer that.